File: database/app/Models/Transaksi.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Transaksi extends Model
{
    use HasFactory;
    public $timestamps = false;

    protected $table = "transaksis";

    protected $primaryKey = "id";

    protected $fillable = [
        'idUser',
        'idPemesananTiket',
        'metodePembayaran',
        'totalHarga',
    ];

    public function users()
    {
        return $this->belongsTo(User::class, 'idUser', 'id');
    }

    public function pemesananTiket()
    {
        return $this->belongsTo(PemesananTiket::class, 'idPemesananTiket', 'id');
    }

    public function history()
    {
        return $this->hasOne(History::class, 'idTransaksi', 'id');
    }

}

File: tubez/lib/entity/transaksi.dart
import 'dart:convert';

class Transaksi {
  final int idUser;
  final int idPemesananTiket;
  final String metodePembayaran;
  final double totalHarga;

  Transaksi({
    required this.idUser,
    required this.idPemesananTiket,
    required this.metodePembayaran,
    required this.totalHarga,
  });

  Map<String, dynamic> toJson() {
    return {
      'idUser': idUser,
      'idPemesananTiket': idPemesananTiket,
      'metodePembayaran': metodePembayaran,
      'totalHarga': totalHarga,
    };
  }
}
